### PR TITLE
Add new generic stats data

### PR DESCRIPTION
**Pull request by jcague**
New data structure to store stats.

### CODE REVIEW

File: erizo/src/erizo/stats/StatNode.h [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  StatNode() {}
+  virtual ~StatNode() {}
+
+  virtual StatNode& operator[](std::string key);
+
+  virtual StatNode& operator[](uint64_t key) { return (*this)[std::to_string(key)]; }
+
+  template <typename Node>
+  void insertStat(std::string key, Node&& stat) {  // NOLINT
+    node_map_.insert(std::make_pair(key, std::make_shared<Node>(std::forward<Node>(stat))));
```
Comment (lodoyun) [inline erizo/src/erizo/stats/StatNode.h]: Can you explain the use of std::forward here? Maybe even leave a comment so we don't break this in the future

File: erizo/src/erizo/stats/StatNode.cpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  checkPeriod();
+}
+
+uint64_t RateStat::value() {
+  checkPeriod();
+  return last_period_calculated_rate_;
+}
+
+std::string RateStat::toString() {
+  checkPeriod();
```
Comment (lodoyun) [inline erizo/src/erizo/stats/StatNode.cpp]: You don't need this as you already do `checkPeriod()` in `value()`

File: erizo/src/erizo/stats/StatNode.cpp [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  return text.str();
+}
+
+StatNode& CumulativeStat::operator=(uint64_t initial) {
+  total_ = initial;
+  return *this;
+}
+
+StatNode CumulativeStat::operator++(int value) {
+  CumulativeStat node{total_};
```
Comment (lodoyun) [inline erizo/src/erizo/stats/StatNode.cpp]: Couldn't you return `*this`? like in `operator=`
Am I missing something?

Comment (jcague) [inline erizo/src/erizo/stats/StatNode.cpp]: It's just the prototype used for the postincrement operator. There's more info here: http://en.cppreference.com/w/cpp/language/operator_incdec

File: erizo/src/erizo/stats/StatNode.h (hunk 1 above)
Comment (jcague) [inline erizo/src/erizo/stats/StatNode.h]: sure! will add a line to remind us its meaning in the future.

File: erizo/src/erizo/stats/StatNode.cpp (hunk 2 above)
Comment (jcague) [inline erizo/src/erizo/stats/StatNode.cpp]: good catch!

**Pull request merged**